File: src/api/commands/window-behavior.ts
import { invokeCommand } from '@/api/client';

export interface WindowBehaviorConfig {
  minimizeAction: 'taskbar' | 'tray';
  closeAction: 'quit' | 'tray';
}

export function getWindowBehaviorConfig() {
  return invokeCommand<WindowBehaviorConfig>('get_window_behavior_config');
}

export function setWindowBehaviorConfig(config: WindowBehaviorConfig) {
  return invokeCommand<void>('set_window_behavior_config', { ...config });
}

File: src-tauri/src/commands/window_behavior.rs
use tauri::State;

use crate::app::state::AppState;
use crate::error::AppError;
use crate::services::window_behavior_service;

#[tauri::command]
pub fn get_window_behavior_config(
    state: State<'_, AppState>,
) -> Result<window_behavior_service::WindowBehaviorConfig, AppError> {
    window_behavior_service::get_config(state.inner())
}

#[tauri::command]
pub fn set_window_behavior_config(
    state: State<'_, AppState>,
    minimize_action: String,
    close_action: String,
) -> Result<(), AppError> {
    window_behavior_service::set_config_values(state.inner(), minimize_action, close_action)
}
